files-community/Files
# GitHub: Added build CI

## Summary

We are finally going to make build & test CI on the GitHub Actions to gain better integrity. This enables us to configure and manage the CI easily, and have better access to the package artifacts.

We had some difficulties below (wrote down to view quickly what we've overcome)
- Not being able to run the interaction tests on GitHub Actions
  - Fixed by adding proper app id, though it is still unclear that we were able to run that tests on Azure DevOps properly.
- Not being able to add a comment on PR that triggered CI
  - Will have fixed by using 'pull_request_target', which gives us more permissions nonethless makes the workflow more secure. If you'd like to know the differences between them, visit GitHub docs.

## Task Checklist

- [x] Added build job
- [x] Added test job
- [x] ~~Added creator of a comment on PR~~

## Known Issues

_N/A_

## PR Checklist

- [x] **Co-authors**: @chingucoding

## Steps To Validate Changes

1. Create a commit on your branch on your files-community fork.
2. Add a PR
3. See CI run building and testing

## Screenshots

_N/A_

[^1]: If the request hasn't been agreed by the team, this work might be rejected. Make sure that you get approval from the team before opening any PR related the request.
[^2]: If you removed any en-US string resources, make sure that there are no references of those resources. When you add a new en-US string resources, do not make any changes on other languages' resources; [Crowdin](https://crowdin.com/project/files-app) will do that, instead.
